src/rasterstats/main.py

```diff
@@ -205,14 +207,10 @@ def gen_zonal_stats(
                 if 'count' in stats:  # special case, zero makes sense here
                     feature_stats['count'] = 0
             else:
+                valid = masked.compressed()
+                keys, counts = np.unique(valid, return_counts=True)
```

Comment:
The unique elements, `keys`, has the really useful property of being sorted. We can take advantage of that for several stats and thus eliminate two O(n) passes (min, max) over the raster data.

Reply:
Nice idea. I like this approach. Two comments:

* We could eliminate the `np.unique` call when the stats do not require it. I guess most stats do now... but we could avoid it entirely if the stats were only  `mean`, `count` or `std`

* Do you have any numbers to show the performance changes pre/post? I'm not seeing much of a positive impact, though my next comment might explain why...

* This impacts memory usage. Keeping the entire keys and counts arrays in memory for the remainder of the loop might be a bad tradeoff in some cases.  Prior to this PR, the `unique` stat was the only one that called `np.unique`, and users had to opt in. That stat is intended for categorical rasters - using it on continuous values would not make much sense. Now we're calling `np.unique` on every invocation and using that to derive other stats - fine in theory. But for continuous rasters or those with 16+ bits per band, there's not likely to be any duplicates - meaning we're allocating a `keys` and `counts` array that are both roughly the length of the masked array itself (possible 200% increase in memory footprint). This causes a performance _regression_ for several use cases; code which never had to call `np.unique` now does, and allocates huge chunks of memory as a result. Multiple O(n) passes have the advantage of not requiring new allocations unless `unique` is specifically requested.

So while this _may_ speed up some use cases, it _may_ slow down others and will certainly cause additional resource usage. I'll have to do some more testing to quantify the impact and evaluate the tradeoffs